lib/vocabularies/code.ts

```diff
@@ -90,8 +92,23 @@ export function callValidateCode(
   return context !== nil ? _`${func}.call(${context}, ${args})` : _`${func}(${args})`
 }
 
-export function usePattern({gen, it: {opts}}: KeywordCxt, pattern: string): Name {
+export function usePattern({gen, it: {self, opts}}: KeywordCxt, pattern: string): Name {
   const u = opts.unicodeRegExp ? "u" : ""
+  const {useRE2} = opts
+  if (u === "u" && useRE2) {
+    try {
+      const engine = new RE2(pattern)
```

Comment:
it's probably incorrect to call it "engine", RE2 is an engine, this one is probably just "regExp" or something...